common/lib/dependabot/git_metadata_fetcher.rb

```diff
@@ -86,6 +90,37 @@ def fetch_raw_upload_pack_for(uri)
       )
     end
 
+    def fetch_raw_upload_pack_with_git_for(uri)
+      status = 200
+
+      service_pack_uri = uri
+      service_pack_uri += ".git" unless service_pack_uri.end_with?(".git")
+
+      command = "git ls-remote #{service_pack_uri}"
+      env = { "PATH" => ENV["PATH"] }
+
+      start = Time.now
+      stdout, stderr, process = Open3.capture3(env, command)
+      time_taken = Time.now - start
+
+      # package the command error like a HTTP response so error handling
+      # remains unchanged
+      unless process.success?
+        status = 500
+        stdout = {
+          message: stderr,
+          command: command,
+          time_taken: time_taken,
+          process_exit_value: process.to_s
+        }
```

Comment:
Please correct me if I'm wrong but it doesn't look like we're using these fields for error processing.

WDYT about simplifying this a bit by something like:

```ruby
    def fetch_raw_upload_pack_with_git_for(uri)
      service_pack_uri = uri
      service_pack_uri += ".git" unless service_pack_uri.end_with?(".git")

      command = "git ls-remote #{service_pack_uri}"
      env = { "PATH" => ENV["PATH"] }

      stdout, stderr, process = Open3.capture3(env, command)
      # package the command error like a HTTP response so error handling
      # remains unchanged
      if process.success?
        OpenStruct.new(body: stdout, status: 200)
      else
        OpenStruct.new(body: stderr, status: 500)
      end
    end
```